### PR TITLE
bpo-46066: catch DeprecationWarning

### PR DESCRIPTION
**Pull request by JelleZijlstra**
https://bugs.python.org/issue46066

### CODE REVIEW

File: Lib/test/test_typing.py [hunk 1]
```diff
@@ -4585,8 +4585,12 @@ def test_typeddict_create_errors(self):
...
         with self.assertRaises(TypeError):
             TypedDict(_typename='Emp', name=str, id=int)
-        with self.assertRaises(TypeError):
-            TypedDict('Emp', _fields={'name': str, 'id': int})
+        
+        # We raise a DeprecationWarning for the keyword syntax
+        # before the TypeError.
+        with self.assertWarns(DeprecationWarning):
+            with self.assertRaises(TypeError):
+                TypedDict('Emp', _fields={'name': str, 'id': int})
```
Comment (sobolevn) [inline Lib/test/test_typing.py]: I think we can remove tests with `_fields`. This field is long gone: https://github.com/python/cpython/commit/8fc5839a9def34c13b6025c291434ba5fb5d6442#diff-ddb987fca5f5df0c9a2f5521ed687919d70bb3d64eaeb8021f98833a2a716887L1723-L1727

Comment (97littleleaf11) [reply Lib/test/test_typing.py]: Agree!

**Review by 97littleleaf11 — COMMENTED** [review]
Oops, sorry that I missed the warnings. Thanks @tirkarthi for pointing out!

File: Lib/test/test_typing.py (hunk 1 above)
Comment (JelleZijlstra) [inline Lib/test/test_typing.py]:
```suggestion
```

Comment (JelleZijlstra) [reply Lib/test/test_typing.py]: Good point, removed it

**Review by sobolevn — APPROVED** [review]
Thank you!

**Comment by JelleZijlstra** [comment]
@gvanrossum Planning to merge this simple PR (fixes an oversight in the TD kwargs deprecation change; approved by Nikita).

**Review by gvanrossum — APPROVED** [review]
I’d rephrase the commit message using “Check deprecation warning”.

**Pull request merged**